app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Enums\AuthenticationStatus;
+use App\Http\Requests\Enum;
+use App\Models\{
+    User,
+    Authentication,
+};
+use App\Http\Requests\UserRequest;
+use Carbon\Carbon;
+use Cookie;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\{
+    Hash,
+    Mail,
+    Validator,
+};
+use Illuminate\Foundation\Auth\RegistersUsers;
+use Illuminate\Auth\Events\Registered;
+use App\Http\Controllers\Controller;
+
+class UserController extends Controller
+{
+    public function create(Request $request)
+    {
+        $token = $request->token;  
+        $authentication = Authentication::where('token', $token)
+            ->where('status', (string)AuthenticationStatus::SEND_MAIL)
+            ->where('expired_at', '>', now())
+            ->first();
+
+        if (!$authentication)
+        {
+            return view('users.invalid_token_error');
+        }
+
+        Cookie::queue('token', $token);
+        return view('users.create');  
+    }
+
+    public function store(UserRequest $request)
+    {
+        $authentication = Authentication::where('token', $request->cookie('token'))->first();
+
+        if (!$authentication) {
+            return view('users.invalid_token_error');
+        }
+
+        User::create([
+            'email'             => $authentication->email,
+            'nickname'          => $request->nickname,
+            'first_name'        => $request->first_name,
+            'last_name'         => $request->last_name,
+            'first_name_kana'   => $request->first_name_kana,
+            'last_name_kana'    => $request->last_name_kana,
+            'gender'            => $request->gender,
+            'postal_code'       => str_replace("-", "", $request->postal_code),
```

Comment:
```suggestion
            'postal_code'       => str_replace('-', '', $request->postal_code),
```
基本的にクォテーションはシングルクォテーション（'）で書いてください。
ダブルクォテーション（"）は文字列を代入した変数を使った変数展開や、正規表現のエスケープに使ったりするので
使い分けましょう。

変数展開や文字列についてはこちらを読んでください。

変数展開
https://qiita.com/tadsan/items/e4796449c736cfb5c9bd#:~:text=%23%3D%3E%20%22Hello%2C%20WORLD!%22-,PHP%E3%81%A7%E3%81%AF,-%E3%81%B5%E3%81%A4%E3%81%86%E3%81%AEPHP

エスケープ
https://www.javadrive.jp/php/string/index4.html